ddtrace/contrib/httplib/patch.py

```diff
@@ -47,17 +54,48 @@ def _wrap_getresponse(func, instance, args, kwargs):
             log.debug('error applying request tags', exc_info=True)
 
 
-def _wrap_putrequest(func, instance, args, kwargs):
+def _wrap_request(func, instance, args, kwargs):
     # Use any attached tracer if available, otherwise use the global tracer
     pin = Pin.get_from(instance)
     if should_skip_request(pin, instance):
         return func(*args, **kwargs)
 
+    cfg = config.get_from(instance)
+
     try:
         # Create a new span and attach to this instance (so we can retrieve/update/close later on the response)
         span = pin.tracer.trace(span_name, span_type=SpanTypes.HTTP)
         setattr(instance, '_datadog_span', span)
 
+        # propagate distributed tracing headers
+        if cfg.get("distributed_tracing"):
+            if len(args) > 3:
+                headers = args[3]
```

Comment:
can the headers also be passed as a `kwarg`?

Reply:
Not if there are more than 3 args.